lighthouse-core/gather/driver.js

```diff
@@ -512,6 +505,35 @@ class Driver {
     });
   }
 
+  /**
+   * Returns LHError if the security state is insecure.
+   * @param {LH.Crdp.Security.SecurityStateChangedEvent} securityState
+   * @returns {LHError|undefined}
+   */
+  checkForSecurityIssues({securityState, explanations}) {
+    if (securityState === 'insecure') {
+      const insecureDescriptions = explanations
+        .filter(exp => exp.securityState === 'insecure')
+        .map(exp => exp.description);
+      return new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {
+        securityMessages: insecureDescriptions.join(' '),
+      });
+    }
+  }
+
+  waitForSecurityIssuesCheck() {
```

Comment:
can we just combine the two of these? Seems unnecessary to split (`waitForSecurityIssuesCheck` doesn't really do anything in isolation and they aren't called separately)

Reply:
we talked earlier about the order of the listener vs calling enable().

......maybe the order right now is OK. if you listened before calling security.enable then perhaps you would get a statechanged event for the about:blank state.

I'm not totally sure about this.

but typically we do like setting up the event listener _before_ enable is called, because enable _usually_ flushes all events before it results.